source/common/http/utility.cc

```diff
@@ -446,5 +446,29 @@ void Utility::transformUpgradeResponseFromH2toH1(HeaderMap& headers, absl::strin
   }
 }
 
+const Router::RouteSpecificFilterConfig*
+Utility::resolvePerFilterConfigGeneric(const std::string& filter_name,
+                                       const Router::RouteConstSharedPtr& route) {
+  if (!route) {
+    return nullptr;
+  }
+
+  const Router::RouteSpecificFilterConfig* maybe_filter_config{};
+
+  const Router::RouteEntry* routeEntry = route->routeEntry();
+  if (routeEntry) {
+    maybe_filter_config = routeEntry->perFilterConfig(filter_name);
+  }
+
+  if (!maybe_filter_config) {
```

Comment:
Nit: prefer explicit comparisons with `nullptr` in the various tests in this method.